cartoframes/data/observatory/enrichment/enrichment_service.py

```diff
@@ -61,37 +61,40 @@ def callback(job):
             raise EnrichmentError(errors)
 
         for df in dfs_enriched:
-            cartodataframe = cartodataframe.merge(df, on=_ENRICHMENT_ID, how='left')
+            geodataframe = geodataframe.merge(df, on=_ENRICHMENT_ID, how='left')
 
         # Remove extra columns
-        cartodataframe.drop(_ENRICHMENT_ID, axis=1, inplace=True)
-        cartodataframe.drop(_GEOM_COLUMN, axis=1, inplace=True)
+        geodataframe.drop(_ENRICHMENT_ID, axis=1, inplace=True)
+        geodataframe.drop(_GEOM_COLUMN, axis=1, inplace=True)
 
-        return cartodataframe
+        return geodataframe
 
     @timelogger
     def _prepare_data(self, dataframe, geom_col):
-        cartodataframe = CartoDataFrame(dataframe, copy=True)
+        geodataframe = GeoDataFrame(dataframe, copy=True)
 
-        if geom_col:
-            cartodataframe.set_geometry(geom_col, inplace=True)
+        if has_geometry(dataframe):
+            geodataframe.set_geometry(dataframe.geometry.name, inplace=True)
```

Comment:
> If the DataFrame has a geometry column (_geometry_column_name), is not it set in the GeoDataFrame instantiation?

There is a bug in geopandas: when the user creates a copy of the GDF from another GDF, both geometry is lost and that's why we need these two lines.

```
GeoDataFrame(geodataframe, copy=True)
```

Note: Since `_geometry_column_name` is a GDF attribute, it's not contained in a pandas.DataFrame instance.

About the refactor, I agree we can compact the code and save calls. Thanks!

Reply:
> There is a bug in geopandas: when the user creates a copy of the GDF from another GDF, both geometry is lost and that's why we need these two lines.

Thank you